### PR TITLE
Adds javadoc to method to document 'magic' behavior

### PR DESCRIPTION
**Pull request by mko-sci**
Fixes: [OX-9538](https://scireum.myjetbrains.com/youtrack/issue/OX-9538)

### CODE REVIEW

**Comment by andyHa** [comment]
Maybe we should also document tne magic parameters errorMessage and errorClass for the i18n version (withFormattedMessage - or the like), while we're at it?

File: src/main/java/sirius/kernel/health/Exceptions.java [hunk 1]
```diff
@@ -114,6 +114,8 @@ protected ErrorHandler(boolean processError) {
 
         /**
          * Specifies which exception leaded to the error being handled.
+         * <p>
+         * Note: When used, an additional '- %s (%s)' has to be added at the end of the error message provided when concatenating a {@link ErrorHandler#withSystemErrorMessage(String, Object...)} call.
```
Comment (mkeckmkeck) [inline src/main/java/sirius/kernel/health/Exceptions.java:118]: Better add this to the withSystemErrorMessage call? Also, outputting the additional parameters is optional.

**Comment by mko-sci** [comment]
Gibts noch mehr stellen wo das Comment net schaden würde?

**Review by sabieber — APPROVED** [review]

**Review by mkeckmkeck — APPROVED** [review]

File: src/main/java/sirius/kernel/health/Exceptions.java [hunk 2]
```diff
@@ -114,6 +114,8 @@ protected ErrorHandler(boolean processError) {
 
         /**
          * Specifies which exception leaded to the error being handled.
+         * <p>
+         * Note: When used, an additional '- %s (%s)' could be added at the end of the error message provided when concatenating a {@link ErrorHandler#withSystemErrorMessage(String, Object...)} call.
```
Comment (andyHa) [inline src/main/java/sirius/kernel/health/Exceptions.java:118]:
```suggestion
         * <p>
         * Note: When {@link ErrorHandler#withSystemErrorMessage(String, Object...)} is used, two extra
         * parameters are available for the string template. The first is the exception message and the second
         * is the exception class. These will be appended after all the user-given parameters. Therefore,
         * one can add <tt>%s (%s)</tt> to the message provided here to output the actual error and
         * exception type if needed.
```

File: src/main/java/sirius/kernel/health/Exceptions.java [hunk 3]
```diff
@@ -167,6 +169,8 @@ public ErrorHandler withDirectMessage(String message) {
          * <p>
          * Still a translated message will be created, which notifies the user about the system error and provides
          * the untranslated error message, generated by this method. These messages should be in english.
+         * <p>
+         * Note: When combining with {@link ErrorHandler#error(Throwable)} an optional '- %s (%s)' could be added at the end of the provided message to output the errorMessage and errorClass.
```
Comment (andyHa) [inline src/main/java/sirius/kernel/health/Exceptions.java:173]:
```suggestion
         * <p>
         * Note: When combining with {@link ErrorHandler#error(Throwable)}, two extra
         * parameters are available for the string template. The first is the exception message and the second
         * is the exception class. These will be appended after all the user-given parameters. Therefore,
         * one can add <tt>%s (%s)</tt> to the message provided here to output the actual error and
         * exception type if needed.
```

**Review by sabieber — APPROVED** [review]

**Review by qw3ry — APPROVED** [review]

**Review by cschierle — APPROVED** [review]

**Pull request merged**